### PR TITLE
Change FixFungibleTokenTotalSupplyMigration to use deduped token balance info

### LINKED ISSUE #8433: Mirrornode returns a total supply of 0 even when some accounts hold the token.
### Description

The mirrornode returns a total_supply of 0 for the testnet USDC token: [link](https://hashscan.io/testnet/token/0.0.429274?p=1), but this is not accurate as some accounts hold the token: [link](https://testnet.mirrornode.hedera.com/api/v1/tokens/0.0.429274/balances).

Maybe important(because of the mirrornode instance), I'm from Europe

### Steps to reproduce

1. Search for this token (0.0.429274) on the Testnet HashScan: https://hashscan.io/testnet/token/0.0.429274?p=1 and check the total supply.
2. Verify that some accounts hold the token: https://testnet.mirrornode.hedera.com/api/v1/tokens/0.0.429274/balances

### Additional context

image

### Hedera network

testnet

### Version

v0.1.0

### Operating system

None

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR fixes the issue which causes incorrect fungible total supply

- Fix the SQL to get token balance info from deduped `token_balance` table
- Update the minimum required version

**Related issue(s)**:

Fixes #8433 

**Notes for reviewer**:

This will only happen on a database with existing data and `FixFungibleTokenTotalSupplyMigration` never ran before.

Thus

- we don't want to increase the checksum and rerun it
- no need to backport

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/FixFungibleTokenTotalSupplyMigration.java [hunk 1]
```diff
@@ -31,34 +31,44 @@ public class FixFungibleTokenTotalSupplyMigration extends RepeatableMigration {
 
     private static final String SQL =
             """
-            with snapshot_timestamp as (
+            with last as (
```
Comment (xin-hedera) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/FixFungibleTokenTotalSupplyMigration.java:34]: The other approach is just sum the balance from associated `token_account` rows, however testing shows it's much slower.

**Review by edwin-greene — APPROVED** [review]
Looks good to me

**Review by jnels124 — APPROVED** [review]
LGTM

**Review by mgoelswirlds — APPROVED** [review]
LGTM

**Pull request merged**